.github/workflows/ci.yml

```diff
@@ -118,6 +118,129 @@ jobs:
             test/functional.test.exe
             test/runhcs.test.exe
             test/sample-logging-driver.exe
+  
+  integration-tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-2019, windows-2022]
+    
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.18.3'
+          check-latest: true
+      - name: Set env
+        shell: bash
+        run: |
+          mkdir -p "${{ github.workspace }}/bin"
+
+          echo "GOPATH=${{ github.workspace }}" >> $GITHUB_ENV
+          echo "${{ github.workspace }}/bin" >> $GITHUB_PATH
+          echo "${{ github.workspace }}/src/github.com/containerd/containerd/bin" >> $GITHUB_PATH
+      - uses: actions/checkout@v2
+        with:
+          path: src/github.com/containerd/containerd
+          repository: 'containerd/containerd'
+
+      - uses: actions/checkout@v2
+        with:
+          path: src/github.com/Microsoft/hcsshim
+
+      - name: Build binaries
+        shell: bash
+        working-directory: src/github.com/containerd/containerd
+        run: |
+          set -o xtrace
+          mingw32-make.exe binaries
+          script/setup/install-cni-windows
+
+      - name: Build the shim
+        working-directory: src/github.com/Microsoft/hcsshim
+        shell: powershell
+        run: |
+          go build -mod vendor -o "${{ github.workspace }}/src/github.com/containerd/containerd/bin/containerd-shim-runhcs-v1.exe" .\cmd\containerd-shim-runhcs-v1
```

Comment:
We upload `containerd-shim-runhcs-v1.exe` as an artifact in the `build` job. Can we use that here instead of rebuilding?

Reply:
We could, but it means tying this job to a different one that may change at some point. The time it takes to do it is short, and I think it's safer to be self contained. If you wish, I can change it to download the artifact.